src/blant.c

```diff
@@ -1379,14 +1381,15 @@ int main(int argc, char *argv[])
 	    if(_confidence >= 1) _confidence /= 100; // user specified percent
 	    break;
 	case 'k': _k = atoi(optarg);
+		_k_base = MIN(_k, OLD_MAX_K);
```

Comment:
Not sure I understand this. So when the requested k on the command line is <=8, then _k_base and _k are the same? I guess maybe that makes sense, but I think it would be better if _k_base is never actually used or needed unless k>8.

Reply:
If I remove _k_base then I need to do some if else on calling the Canon map reading, Permutation reading  functions

I thought It would be better to just have _k_base which will be equal to _k for k <= 8

Reply:
I can do that if you want

Reply:
OK I get it. I think it's OK for now.